### PR TITLE
Gb 16

### PR DESCRIPTION
**Pull request by veranika107**

### CODE REVIEW

**Comment by veranika107** [comment]
[GB - Implement social auth via google connect](https://trello.com/c/YQwg8Ukh/16-gb-implement-social-auth-via-google-connect)

File: src/Security/GoogleAuthenticator.php [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        return $request->attributes->get('_route') === 'connect_google_check';
+    }
+
+    public function authenticate(Request $request): Passport
+    {
+        /** @var GoogleClient $client */
+        $client = $this->clientRegistry->getClient('google');
+        $accessToken = $this->fetchAccessToken($client);
+
+        return new SelfValidatingPassport(
```
Comment (wechka53) [inline src/Security/GoogleAuthenticator.php:41]: add try catch and re throw exception

File: src/Controller/GoogleConnectController.php [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+use KnpU\OAuth2ClientBundle\Client\ClientRegistry;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Routing\Annotation\Route;
+
+class GoogleConnectController extends AbstractController
+{
+    #[Route('/connect/google', name: 'connect_google_start')]
+    public function connectGoogle(ClientRegistry $clientRegistry)
```
Comment (wechka53) [inline src/Controller/GoogleConnectController.php:13]: add tests

**Review by wechka53 — CHANGES_REQUESTED** [review]

File: config/packages/security.yaml [hunk 3]
```diff
@@ -7,22 +7,26 @@ security:
         # used to reload user from session & other features (e.g. switch_user)
         app_user_provider:
             entity:
-                class: App\Entity\Admin
-                property: username
```
[the remark this answers is not in the document]
Comment (veranika107) [reply config/packages/security.yaml:11]: thank you, fixed

File: src/Security/GoogleAuthenticator.php (hunk 1 above)
Comment (veranika107) [reply src/Security/GoogleAuthenticator.php:39]: as we discussed, exceptions are thrown in other method. Sentry is added instead

File: src/Controller/GoogleConnectController.php (hunk 2 above)
Comment (veranika107) [reply src/Controller/GoogleConnectController.php:13]: done, thanks

**Review by wechka53 — APPROVED** [review]

**Pull request merged**